### ISSUE TITLE
Replace DuckDB Parquet writer to Arrow writer for ParquetTpchTest 

### ISSUE THREAD

**Issue by yingsu00** [facebookincubator/velox#2346]
Parquet spec requires each ColumnChunk's file_offset to be set correctly.  This is the Parquet cpp generated from thrift definition:
```
uint32_t ColumnChunk::read(::apache::thrift::protocol::TProtocol* iprot) {
  ::apache::thrift::protocol::TInputRecursionTracker tracker(*iprot);
  ...
  bool isset_file_offset = false;

  while (true) {
    xfer += iprot->readFieldBegin(fname, ftype, fid);
    if (ftype == ::apache::thrift::protocol::T_STOP) {
      break;
    }
    switch (fid) {
      case 1:
        ...
        break;
      case 2:
        if (ftype == ::apache::thrift::protocol::T_I64) {
          xfer += iprot->readI64(this->file_offset);      // This is 0 for all columns
          isset_file_offset = true;                                   // But isset_file_offset becomes true
        } else {
          xfer += iprot->skip(ftype);
        }
        break;
      ....

  if (!isset_file_offset)
    throw TProtocolException(TProtocolException::INVALID_DATA);    // If isset_file_offset is not set the file is corrupted
  return xfer;
}
```
By specification, the file_offset of the RowGroup is optional. However DuckDB writes it for all RowGroups but does NOT write the required file_offset of the ColumnChunk. Take Q3 for example, the metadata being parsed is in the following picture:
The table has 1 rowgroups and 5 columns. All 5 columns have file_offset = 0, but the row_group's file_offset is 4.
<img width="1173" alt="Screen Shot 2022-08-20 at 4 17 04 AM" src="https://user-images.githubusercontent.com/33299678/185743506-1ae38ff5-32ff-4afe-9968-06862abf853e.png">

This may result reading wrong rowgroups if we were not checking both. This test uses TINY scale factor and only contains 1 RowGroup therefore the RowGroup skipping was not tested. 

Another issue is DuckDB does not do Dictionaries encoding properly. For TPCH, dictionary encoding is crucial. 

Son summary we may want to 
1) change the Parquet file writer from DuckDB to Arrow. 
2) Add RowGroup elimination coverage to this test. 
The 2) is not urgent.

**Comment by yingsu00** [comment]
cc @oerling @majetideepak

**Comment by majetideepak** [comment]
@yingsu00 Please add tests to https://github.com/facebookincubator/velox/blob/main/velox/dwio/parquet/tests/reader/E2EFilterTest.cpp if you want to add tests with dictionary encoding and to use the Arrow writer. There are already some tests there with dictionary encoding.

The ParquetTpchTest is currently passing. Is there an issue with the test setup that is hiding the issue you posted? Do you have a reproducer?

**Comment by yingsu00** [comment]
@majetideepak I actually made a mistake in describing the issue. There was no failure for this particular case and sorry for the confusion. This issue is mainly for ParquetTpchTest to use Arrow Parquet writer. DuckDB writes RG file_offset but not ColumnChunk file_offset which is different from the spec. Another hope was to make this test cover the multiple RG case. E2EFilterTest already covers multiple RowGroup tests. I've updated the description accordingly.
